File: hashmap/group-anagrams/group-anagrams.go
func groupAnagrams(strs []string) [][]string {
    m := make(map[string][]string)
	for _, s := range strs {
		key := sortString(s)
		m[key] = append(m[key], s)
	}
	res := make([][]string, 0, len(m))
	for _, v := range m {
		res = append(res, v)
	}
	return res
}

func sortString(s string) string {
	r := []rune(s)
	sort.Slice(r, func(i, j int) bool { return r[i] < r[j] })
	return string(r)
}

File: hashmap/group-anagrams/group-anagrams.cs
public class Solution
{
    public IList<IList<string>> GroupAnagrams(string[] strs)
    {
        var map = new Dictionary<string, List<string>>();

        foreach (var s in strs)
        {
            var chars = s.ToCharArray();
            Array.Sort(chars);
            string key = new string(chars);
            if (!map.ContainsKey(key)) map[key] = new List<string>();
            map[key].Add(s);
        }

        return map.Values.ToList<IList<string>>();
    }
}